source/extensions/filters/http/ext_proc/processing_request_modifier.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/config/typed_config.h"
+#include "envoy/server/factory_context.h"
+#include "envoy/service/ext_proc/v3/external_processor.pb.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/http/ext_proc/matching_utils.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+// Interface to modify the processing request before sending it to the ext_proc backend
+class ProcessingRequestModifier {
+public:
+  virtual ~ProcessingRequestModifier() = default;
+
+  struct Params {
+    envoy::config::core::v3::TrafficDirection traffic_direction;
+    Http::StreamFilterCallbacks* callbacks;
+    const Http::RequestHeaderMap* request_headers;
+    const Http::RequestOrResponseHeaderMap* response_headers;
+    const Http::HeaderMap* response_trailers;
+  };
+
+  // Called to modify the request. Takes in the params needed to create the attributes.
+  // Implementations are expected to set attributes and then return true iff any were set.
```

Comment:
I think implementation may do more than that. They will actually be able modify any element in the `ProcessingRequest` element.